cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/services/ApplicationLifecycleService.java

```diff
@@ -315,10 +316,30 @@ public ApplicationDetail getAppDetail(ApplicationId appId) throws Exception {
     if (appSpec == null) {
       throw new ApplicationNotFoundException(appId);
     }
+    // TODO: fetch creation time from store
     String ownerPrincipal = ownerAdmin.getOwnerPrincipal(appId);
     return enforceApplicationDetailAccess(appId, ApplicationDetail.fromSpec(appSpec, ownerPrincipal));
   }
 
+  /**
+   * Get details about the latest version of the specified application
+   *
+   * @param appId the id of the application to get
+   * @return detail about the latest version of the specified application
+   * @throws ApplicationNotFoundException if the specified application does not exist
+   */
+  public ApplicationDetail getAppLatestVersionDetail(ApplicationId appId) throws Exception {
```

Comment:
ApplicationId already has a version, this should just take NamespaceId and app name

Reply:
also less verbose to name this getLatest() or getLatestApp()